### PR TITLE
Add support for other clouds (pt. 2)

### PR DESCRIPTION
**Pull request by danieljurek**
Building atop the options refactor.

### CODE REVIEW

File: cli/azd/pkg/auth/manager.go
Comment (danieljurek) [inline cli/azd/pkg/auth/manager.go:384]: @weikanglim should we use the options object here or the one defined in line 114? The object-level options does involve a cache.

File: cli/azd/pkg/cloud/cloud.go
Comment (weikanglim) [inline cli/azd/pkg/cloud/cloud.go:30]: Is there a way to get this via azure SDK?

File: cli/azd/pkg/cloud/cloud.go
Comment (weikanglim) [inline cli/azd/pkg/cloud/cloud.go:1]: We'll have to leave this up for voting, but I prefer if we can start sticking to a single `azure` package. I'm not sure if this causes cyclic dependencies currently because of how splayed the azure clients are currently

File: cli/azd/pkg/cloud/cloud.go
Comment (danieljurek) [reply cli/azd/pkg/cloud/cloud.go:30]: Storage blob clients in the Azure SDK for Go all require a URL or, in the case of a connection string, an account name and an endpoint suffix (e.g. core.windows.net). In the case of the suffix, the default suffix for Azure Public is hardcoded... but there isn't a direct listing of endpoint suffixes available in the SDK.

File: cli/azd/pkg/cloud/cloud.go
Comment (danieljurek) [reply cli/azd/pkg/cloud/cloud.go:1]: Switching cloud.go to `azure` and updating references does not result in an import cycle. This is likely preferable given the relative simplicity of cloud.

File: cli/azd/pkg/auth/manager.go
Comment (danieljurek) [inline cli/azd/pkg/auth/manager.go:232]: @ellismg, @chlowell -- Did I add in cloud correctly here?

**Review by vhvb1989 — APPROVED** [review]

File: schemas/v1.0/azure.yaml.json
Comment (wbreza) [inline schemas/v1.0/azure.yaml.json:537]: Since we don't have a good release story for our schemas consider holding off on this and putting it only in beta for now and then do a smaller PR on release day where we update the v1.0 schema.

File: templates/common/infra/bicep/core/host/container-app.bicep
Comment (wbreza) [inline templates/common/infra/bicep/core/host/container-app.bicep:32]: I'd recommend we split out all the `template` updates into a following PR after the bits are merged in.  I don't recall having a good experience validating template changes that require code changes.

**Review by wbreza — COMMENTED** [review]
Looking great over all.  Main feedback:

1. Split out template updates from code updates into a separate PR that follows the code changes.
2. Split out v1.0 schema updates to their own small PR since we don't have a good release strategy for schemas.

File: cli/azd/cmd/auth_token_test.go
Comment (danieljurek) [reply cli/azd/cmd/auth_token_test.go:27]: It is intentional https://learn.microsoft.com/en-us/entra/identity-platform/scopes-oidc#trailing-slash-and-default

> In this case, a trailing slash on the resource URI means the slash must be present when the token is requested. So when you request a token for https://management.azure.com/ and use .default, you must request https://management.azure.com//.default (notice the double slash!).

File: cli/azd/pkg/project/container_helper.go
Comment (danieljurek) [reply cli/azd/pkg/project/container_helper.go:191]: Injecting the configured cloud object is more efficient.

File: schemas/v1.0/azure.yaml.json
Comment (danieljurek) [reply schemas/v1.0/azure.yaml.json]: https://github.com/Azure/azure-dev/pull/3482

File: templates/common/infra/bicep/core/host/container-app.bicep
Comment (danieljurek) [reply templates/common/infra/bicep/core/host/container-app.bicep]: https://github.com/Azure/azure-dev/pull/3481

File: cli/azd/cmd/container.go
Comment (danieljurek) [reply cli/azd/cmd/container.go:392]: Reversing and short-circuiting is a better approach as it requires reading fewer configs. I'll do that.

File: cli/azd/pkg/cloud/cloud.go
Comment (danieljurek) [reply cli/azd/pkg/cloud/cloud.go:1]: We can refactor this later. It's possible and fairly straightforward.

**Review by weikanglim — APPROVED** [review]

**Review by wbreza — APPROVED** [review]

File: cli/azd/pkg/auth/manager.go
Comment (chlowell) [inline cli/azd/pkg/auth/manager.go:450]: If you just want to mark requests as originating from azd, you can set `ClientOptions.Telemetry.ApplicationID`. That gets a value like `<your text here> azsdk-go-azidentity/v1.6.0-beta.2 (go1.21.6; linux)`

Comment (danieljurek) [reply cli/azd/pkg/auth/manager.go:465]: Adding this work item to handle user agent scenarios https://github.com/Azure/azure-dev/issues/2723

File: cli/azd/pkg/auth/manager.go
Comment (danieljurek) [reply cli/azd/pkg/auth/manager.go:409]: This can be handled as part of making sure the user agent is consistent: https://github.com/Azure/azure-dev/issues/2723

**Pull request merged**
